src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs

```diff
@@ -337,16 +337,117 @@ public Utf8String Substring(int index, int length)
             }
         }
 
-        public Utf8String SubstringFrom(Utf8String substring)
+        // TODO: Should this be public or should focused on scenarios?
+        private int IndexOf(Utf8String substring)
         {
             throw new NotImplementedException();
         }
 
-        public Utf8String SubstringTo(Utf8String substring)
+        // TODO: Should this be public or should focused on scenarios?
+        private int IndexOf(Utf8CodeUnit codeUnit)
+        {
+            for (int i = 0; i < Length; i++)
+            {
+                if (codeUnit == GetCodeUnitAtPositionUnchecked(i))
+                {
+                    return i;
+                }
+            }
+
+            return -1;
+        }
+
+        // TODO: Should this be public or should focused on scenarios?
+        private int IndexOf(UnicodeCodePoint codePoint)
         {
             throw new NotImplementedException();
         }
 
+        public bool TrySubstringFrom(Utf8String substring, out Utf8String result)
```

Comment:
To me the result is the substring. The substring parameter is more like a prefix the method will be searching for. But please feel to commit this change, just make a note so we talk about this name at the API review.